image-upload

```diff
@@ -42,9 +42,9 @@ def upload(dest, source, public):
     if s3.is_key_present(url):
         headers = {s3.ACL: s3.PUBLIC} if public else {}
 
-        # slightly magic: image filenames always end with -{sha256}.qcow2
-        assert source.endswith('.qcow2')
-        hash_value = source[-70:-6]
+        # slightly magic: image filenames always end with -{sha256}.qcow2 or -{sha256}.iso
+        assert (source.endswith('.qcow2') or source.endswith('.iso'))
+        hash_value = source.rstrip(".qcow2").rstrip(".iso")[-64:]
```

Comment:
Hmm, but that now includes the initial file name ("fedora-rawhide-boot") into the hash, which we don't want. There was a reason for the 64 character slice :grin:

Reply:
Argh, missed the `[-64:]`, sorry. And the upload obviously worked. So please push the commit with the rebuild, then it's ready to go.

Reply:
I thought the hash value is meant to contain for a `source` like +fedora-rawhide-boot-0b2fd35c87c4fc7f38149bb2a7675e3dcddc8f9c9da7bb739f02d5650359814a.iso exactly the 0b2fd35c87c4fc7f38149bb2a7675e3dcddc8f9c9da7bb739f02d5650359814a part.

```
[kkoukiou@sequioa bots]$ ipython 
Python 3.10.2 (main, Jan 17 2022, 00:00:00) [GCC 11.2.1 20211203 (Red Hat 11.2.1-7)]
Type 'copyright', 'credits' or 'license' for more information
IPython 7.26.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: source="fedora-rawhide-boot-0b2fd35c87c4fc7f38149bb2a7675e3dcddc8f9c9da7bb739f02d5650359814a.iso"

In [2]: source.rstrip(".qcow2").rstrip(".iso")[-64:]
Out[2]: '0b2fd35c87c4fc7f38149bb2a7675e3dcddc8f9c9da7bb739f02d5650359814a'
```